source/common/network/dns_impl.cc

```diff
@@ -66,11 +66,18 @@ absl::optional<std::string> DnsResolverImpl::maybeBuildResolversCsv(
 DnsResolverImpl::AresOptions DnsResolverImpl::defaultAresOptions() {
   AresOptions options{};
 
-  if (use_tcp_for_dns_lookups_) {
+  if (dns_resolver_options_.has_use_tcp_for_dns_lookups() &&
+      dns_resolver_options_.use_tcp_for_dns_lookups().value()) {
```

Comment:
Is there a way to eliminate the usage of `has_use_tcp_for_dns_lookups()`?

Reply:
I verified this. The has_** is redundant here. I removed them.